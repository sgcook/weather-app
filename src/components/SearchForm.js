import React, { useEffect } from "react";
import PropTypes from "prop-types";
import "../styles/SearchForm.css";

function SearchForm({ searchText, setSearchText, onSubmit }) {
  const handleInputChange = (event) => setSearchText(event.target.value);

  useEffect(() => {
    const listener = (e) => {
      if (e.code === "Enter" || e.code === "NumpadEnter") {
        e.preventDefault();
        onSubmit();
      }
    };

    document.addEventListener("keydown", listener);
    return () => {
      document.removeEventListener("keydown", listener);
    };
  }, [onSubmit]);

  return (
    <div className="search-form">
      <label htmlFor="search">
        <input
          type="text"
          placeholder="Search City"
          onChange={handleInputChange}
          value={searchText.toUpperCase()}
        />
      </label>
      <button type="submit" onClick={onSubmit} className="search-button">
        Search
      </button>
    </div>
  );
}

SearchForm.propTypes = {
  searchText: PropTypes.string.isRequired,
  setSearchText: PropTypes.func.isRequired,
  onSubmit: PropTypes.func.isRequired,
};

export default SearchForm;
