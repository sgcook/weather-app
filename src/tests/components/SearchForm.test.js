import React from "react";
import { render } from "@testing-library/react";
import SearchForm from "../../components/SearchForm";

describe("Search Form", () => {
  const validProps = {
    searchText: "London, UK",
    setSearchText: jest.fn,
    onSubmit: jest.fn,
  };

  it("renders correctly", () => {
    const { asFragment } = render(
      <SearchForm
        searchText={validProps.searchText}
        setSearchText={validProps.setSearchText}
        onSubmit={validProps.onSumbit}
      />
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it("button displays correct text", () => {
    const { getByText } = render(
      <SearchForm
        searchText={validProps.searchText}
        setSearchText={validProps.setSearchText}
        onSubmit={validProps.onSumbit}
      />
    );

    expect(getByText("Search")).toBeInstanceOf(HTMLButtonElement);
  });
});
