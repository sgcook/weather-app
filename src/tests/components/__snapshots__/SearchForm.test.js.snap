// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Form renders correctly 1`] = `
<DocumentFragment>
  <div
    class="search-form"
  >
    <label
      for="search"
    >
      <input
        placeholder="Search City"
        type="text"
        value="LONDON, UK"
      />
    </label>
    <button
      class="search-button"
      type="submit"
    >
      Search
    </button>
  </div>
</DocumentFragment>
`;
